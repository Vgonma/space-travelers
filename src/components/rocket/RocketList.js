import React from 'react';
import { useSelector } from 'react-redux';
import RocketItem from './RocketItem';

const RocketList = () => {
  const rockets = useSelector((state) => state.rootReducer.rocketReducer);

  return (
    <div>
      {rockets.map((rocket) => (
        <div key={rocket.id}>
          <RocketItem
            id={rocket.id}
            rocketName={rocket.rocketName}
            description={rocket.description}
            image={rocket.images[0]}
          />
        </div>
      ))}
    </div>
  );
};

export default RocketList;
