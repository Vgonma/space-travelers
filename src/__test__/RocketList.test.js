import React from 'react';
import { render } from '@testing-library/react';
import { useSelector } from 'react-redux';
import RocketList from '../components/rocket/RocketList';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

describe('RocketList Component', () => {
  let tree;

  beforeEach(() => {
    const rocketList = [{
      id: 1, rocketName: 'Falcon 9', description: 'A two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit.', images: [], reserved: false,
    }];

    useSelector.mockImplementation(() => rocketList);

    tree = render(<RocketList />);
  });

  test('should render correctly', () => expect(tree).toMatchSnapshot());
});
