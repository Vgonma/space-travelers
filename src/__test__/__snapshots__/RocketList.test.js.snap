// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketList Component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rocket_container"
      >
        <div>
          <li
            class="rocket"
          >
            <img
              alt="rocket"
            />
            <div
              class="single-rocket-container"
            >
              <span
                class="rocket-name"
                data-testid="name"
              >
                Falcon 9
              </span>
              <div
                class="rocket-detail"
              >
                <span
                  class=""
                />
                A two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit.
              </div>
              <button
                class="reserve-rocket"
                data-testid="button"
                type="button"
              >
                Reserve Rocket
              </button>
            </div>
          </li>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rocket_container"
    >
      <div>
        <li
          class="rocket"
        >
          <img
            alt="rocket"
          />
          <div
            class="single-rocket-container"
          >
            <span
              class="rocket-name"
              data-testid="name"
            >
              Falcon 9
            </span>
            <div
              class="rocket-detail"
            >
              <span
                class=""
              />
              A two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit.
            </div>
            <button
              class="reserve-rocket"
              data-testid="button"
              type="button"
            >
              Reserve Rocket
            </button>
          </div>
        </li>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
